### DISCUSSION TITLE
is_nsfw does not work

### DISCUSSION THREAD

**Discussion by nonetrix** [Rapptz/discord.py#5933] [Q&A] [answered]
trying to make a command for my discord server that only works in NSFW channels but when I add this line:

`if ctx.channel.is_nsfw():`

it returns:

`discord.ext.commands.errors.CommandInvokeError: Command raised an exception: AttributeError: 'DMChannel' object has no attribute 'is_nsfw'`

it seems it's been removed for some odd reason I heard a lot about discord.py rewirte but it's all 404's and the project seems to be dead I would like a workaround or something like that if possible

**Comment by AbstractUmbra** [comment, marked as answer]
'discord.py rewrite' is not dead, but discord.py has not been referred to as this for many years now. You're actually within the repo right now and our doc can be found [here](https://discordpy.readthedocs.io/en/latest/).

Currently we're on version 1.5, known as latest if you'd prefer a name for it.

DMChannels do not have an NSFW filter on them, and as such this check will be invalid. You should check for the channel being a non-Guild channel before doing this to prevent this error.

**Reply by nonetrix** [reply]
> 'discord.py rewrite' is not dead, but discord.py has not been referred to as this for many years now. You're actually within the repo right now and our doc can be found [here](https://discordpy.readthedocs.io/en/latest/).
> 
> Currently we're on version 1.5, known as latest if you'd prefer a name for it.
> 
> DMChannels do not have an NSFW filter on them, and as such this check will be invalid. You should check for the channel being a non-Guild channel before doing this to prevent this error.

I'm dumb oof also random question will checking just add extra bloat if the error is harmless I'm not a expert so I don't know the best way to avoid bloat lol
